src/types/types.d.ts

```diff
@@ -9,6 +9,8 @@ type PriceDetail = {
   value: number;
 };
 
+type SetSnackbar = Dispatch<SetStateAction<boolean>>;
```

Comment:
keep the types enclosed with their component, something like this (Missing types in code snippet)

```
export type SetSnackbar = Dispatch<SetStateAction<boolean>>;

const SnackbarWrapper = ({ children }) => {
  const [open, setOpen] = useState(false);
  const setSnackbar = (state: boolean) => {
    setOpen(state);
  };

  <>
    {children}
    <Snackbar
      className="snackbar" // TODO: Change globals for styles.snackbar
      autoHideDuration={3000}
      open={open}
      onClose={() => setSnackbar(false)}
      message="Añadido al carrito"
      TransitionComponent={Slide}
    ></Snackbar>
  </>
  
};

export const withSnackbar = (Component) => (props) => (
  <SnackbarWrapper>
    <Component {...props} />
  </SnackbarWrapper>
);
```

However, you don't want to have all your types centralized in a file, `src/types` can be just for business-logic related types, such as Product, User, idk

But a Snackbar is a general purpose component